crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1234,19 +1235,26 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
     /// Tables updated:
     ///  * consensus_message_processed - indicate that this certificate was processed by consensus
     ///  * last_consensus_index - records last processed position in consensus stream
+    ///  * consensus_message_order - records at what position this transaction was first seen in consensus
     /// Self::consensus_message_processed returns true after this call for given certificate
     fn finish_consensus_message_process(
         &self,
-        mut batch: DBBatch,
+        batch: DBBatch,
         certificate: &VerifiedCertificate,
         consensus_index: ExecutionIndicesWithHash,
     ) -> SuiResult {
         let transaction_digest = *certificate.digest();
-        let index_to_write = iter::once((LAST_CONSENSUS_INDEX_ADDR, consensus_index));
-        batch = batch.insert_batch(&self.epoch_tables().last_consensus_index, index_to_write)?;
-        batch = batch.insert_batch(
+        let batch = batch.insert_batch(
+            &self.epoch_tables().consensus_message_order,
+            [(consensus_index.index.clone(), transaction_digest)],
+        )?;
+        let batch = batch.insert_batch(
+            &self.epoch_tables().last_consensus_index,
+            [(LAST_CONSENSUS_INDEX_ADDR, consensus_index)],
```

Comment:
is this not just the same as the the highest consensus index in the table? do we need to store this separately?

Reply:
Lol, good catch @mystenmark !

Reply:
Actually looking more, unfortunately we need to keep this separate table, because it contains last ExecutionIndex *and* consensus integrity hash

Reply:
ah, right.